### PR TITLE
Add missing features for HTMLLegendElement

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR is a part of a project to add missing interfaces and interface features to BCD that are from a spec in Editor's Draft or more and is supported in at least one browser.  This particular PR adds missing features, populating the results using data from the [mdn-bcd-collector](https://mdn-bcd-collector.appspot.com) project (v1.1.6), for the HTMLLegendElement API.

Spec: https://html.spec.whatwg.org/multipage/semantics.html#htmllegendelement
IDL: https://github.com/w3c/webref/blob/master/ed/idl/html.idl

### CODE REVIEW

File: api/HTMLLegendElement.json [hunk 1]
```diff
@@ -46,6 +46,100 @@
...
+              "version_added": "1.0"
+            },
+            "webview_android": {
+              "version_added": "1"
+            }
+          },
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
```
Comment (foolip) [inline api/HTMLLegendElement.json:93]:
```suggestion
            "deprecated": true
```

https://html.spec.whatwg.org/multipage/obsolete.html#HTMLLegendElement-partial

File: api/HTMLLegendElement.json [hunk 2]
```diff
@@ -46,6 +46,100 @@
           "standard_track": true,
           "deprecated": false
         }
+      },
+      "align": {
```
Comment (foolip) [inline api/HTMLLegendElement.json:50]: Can you add the corresponding entry to legend.json too?

File: html/elements/legend.json [hunk 3]
```diff
@@ -24,29 +24,123 @@
...
+              }
+            },
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
+        },
+        "form": {
```
Comment (foolip) [inline html/elements/legend.json:98]: So it looks like `<legend form>` isn't a content attribute. Try validating this markup:
```html
<!DOCTYPE html>
<html>
<head>
<title>Test</title>
</head>
<body>
<p></p>
<form id=bla></form>
<fieldset form=bla>
<select form=bla></select>
<label form=bla></label>
</fieldset>
</body>
</html>
```

The form attribute isn't allowed on `<legend>`. The spec is complicated, but from a quick read I think it makes sense:

https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#form-owner

**Review by foolip — COMMENTED** [review]
I think the `form` content attribute isn't a thing. But looking into it also made me unsure about `align`. @sideshowbarker if you have the time, could you give this a look over?

File: html/elements/legend.json [hunk 4]
```diff
@@ -24,29 +24,123 @@
...
+                "version_added": "1"
+              }
+            },
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
         }
```
Comment (sideshowbarker) [inline html/elements/legend.json:144]:
```suggestion
```

**Comment by sideshowbarker** [comment]
> I think the `form` content attribute isn't a thing. But looking into it also made me unsure about `align`. @sideshowbarker if you have the time, could you give this a look over?

https://html.spec.whatwg.org/multipage/obsolete.html#attr-legend-align defines the `align` content/markup attribute for `legend` as on obsolete, non-conforming feature. So it’s appropriate for us to mark it in BCD as deprecated.

File: html/elements/legend.json [hunk 5]
```diff
@@ -24,30 +24,77 @@
...
+                "version_added": "1"
+              }
+            },
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": true
+            }
+          }
+        },
```
Comment (foolip) [inline html/elements/legend.json:97]:
```suggestion
        }
```

File: html/elements/legend.json [hunk 6]
```diff
@@ -24,29 +24,76 @@
...
+              "version_added": "1"
             }
           },
           "status": {
             "experimental": false,
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "align": {
```
Comment (foolip) [inline html/elements/legend.json:51]: OK, was also able to verify that `<legend align=middle>foo</legend>` does center the text, so the content attribute does do something. (I suspected that maybe it was merely reflected but had no effect, but Chromium code search helped me figure it out.)

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
Thanks for helping verify the `<legend form>` situation, @sideshowbarker!

**Pull request merged**